src/saddle.ts

```diff
@@ -23,6 +23,17 @@ export interface Saddle {
   trace: (receipt: TransactionReceipt, options: TraceOptions) => Promise<any>
 }
 
+function allowUndefinedArgs(args: any[] | SendOptions, sendOptions?: SendOptions): [any[], SendOptions | undefined] {
+  if (!Array.isArray(args)) {
+    if (sendOptions !== undefined) {
+      throw new Error(`Args expected to be an array, got ${args}`);
+    }
+    return [[], args];
```

Comment:
I guess in this case args is actually sendOptions?

Reply:
Yeah, typescript isn't great with "you can skip an arg" while JavaScript loves it. Makes for messy code like this.